pandas/core/indexes/period.py

```diff
@@ -683,7 +692,19 @@ def to_timestamp(self, freq=None, how='start'):
         base, mult = _gfc(freq)
         new_data = self.asfreq(freq, how)
 
-        new_data = period.periodarr_to_dt64arr(new_data._ndarray_values, base)
+        end = how == 'E'
+        if end:
+            indexer = np.where(new_data.notnull())
+            # move forward one period
+            new_data._values[indexer] += 1
+            ndarray_vals = new_data._ndarray_values
```

Comment:
I think this should be pushed to periodarr_to_dt64arr, IOW it should take a how=S/E arg